File: backend/src/main/java/edu/cit/sapatosan/entity/OrderEntity.java
package edu.cit.sapatosan.entity;

import java.util.Date;
import java.util.List;

public class OrderEntity {
    public enum PaymentStatus {
        PENDING, PAID, UNPAID
    }

    private String id;
    private String userId;
    private Date orderDate;
    private Double totalAmount;
    private String status;
    private List<String> orderProductIds;
    private String paymentId;
    private PaymentStatus paymentStatus;

    private String firstName;
    private String lastName;
    private String email;
    private String address;
    private String postalCode;
    private String country;
    private String contactNumber;

    public OrderEntity() {}

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public Date getOrderDate() { return orderDate; }
    public void setOrderDate(Date orderDate) { this.orderDate = orderDate; }

    public Double getTotalAmount() { return totalAmount; }
    public void setTotalAmount(Double totalAmount) { this.totalAmount = totalAmount; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public List<String> getOrderProductIds() { return orderProductIds; }
    public void setOrderProductIds(List<String> orderProductIds) { this.orderProductIds = orderProductIds; }

    public String getPaymentId() { return paymentId; }
    public void setPaymentId(String paymentId) { this.paymentId = paymentId; }

    public PaymentStatus getPaymentStatus() { return paymentStatus; }
    public void setPaymentStatus(PaymentStatus paymentStatus) { this.paymentStatus = paymentStatus; }

    public String getFirstName() { return firstName; }
    public void setFirstName(String firstName) { this.firstName = firstName; }

    public String getLastName() { return lastName; }
    public void setLastName(String lastName) { this.lastName = lastName; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public String getPostalCode() { return postalCode; }
    public void setPostalCode(String postalCode) { this.postalCode = postalCode; }

    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; }

    public String getContactNumber() { return contactNumber; }
    public void setContactNumber(String contactNumber) { this.contactNumber = contactNumber; }
}

File: frontend_mobile/app/src/main/java/com/frontend_mobile/models/OrderEntity.kt
package com.frontend_mobile.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.util.Date

@Parcelize
data class OrderEntity(
    var id: String? = null,
    var userId: String? = null,
    var orderDate: Date? = null,
    var totalAmount: Double? = null,
    var status: String? = null,
    var orderProductIds: List<String>? = null,
    var paymentId: String? = null,
    var paymentStatus: PaymentStatus? = null,
    var firstName: String? = null,
    var lastName: String? = null,
    var email: String? = null,
    var address: String? = null,
    var postalCode: String? = null,
    var country: String? = null,
    var contactNumber: String? = null,
): Parcelable

enum class PaymentStatus{
    PENDING, PAID, UNPAID
}